libs/utils/src/lsn.rs

```diff
@@ -343,4 +414,64 @@ mod tests {
         assert_eq!(lsn.fetch_max(Lsn(6000)), Lsn(5678));
         assert_eq!(lsn.fetch_max(Lsn(5000)), Lsn(6000));
     }
+
+    #[test]
+    fn test_lsn_serde() {
+        let lsn = Lsn(1234);
+
+        // Testing human_readable ser/de
+        let serializer = Serializer::builder().is_human_readable(false).build();
+        assert_ok_eq!(lsn.serialize(&serializer), Tokens(vec![Token::U64(1234)]));
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(false)
+            .tokens(Tokens(vec![Token::U64(1234)]))
+            .build();
+        assert_ok_eq!(Lsn::deserialize(&mut deserializer), lsn);
+
+        // Testing NON human_readable ser/de
+        let serializer = Serializer::builder().is_human_readable(true).build();
+        assert_ok_eq!(
+            lsn.serialize(&serializer),
+            Tokens(vec![Token::Str(String::from("0/4D2"))])
+        );
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(true)
+            .tokens(Tokens(vec![Token::Str(String::from("0/4D2"))]))
+            .build();
+        assert_ok_eq!(Lsn::deserialize(&mut deserializer), lsn);
+
+        // Testing mismatching ser/de
+        let serializer = Serializer::builder().is_human_readable(false).build();
+        let f_serialized = lsn.serialize(&serializer).unwrap();
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(true)
+            .tokens(f_serialized)
+            .build();
+        assert!(Lsn::deserialize(&mut deserializer).is_err());
+
+        let serializer = Serializer::builder().is_human_readable(true).build();
+        let t_serialized = lsn.serialize(&serializer).unwrap();
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(false)
+            .tokens(t_serialized)
+            .build();
+        assert!(Lsn::deserialize(&mut deserializer).is_err());
+    }
```

Comment:
Instead of fully generic serialization test case I would like to see actual bincode tests of types containing Lsn be serialized to bytes, and those bytes be deserialized back to the equal structs as in using `assert_eq!`. For serde_json we have such tests with IndexPart at least.

But... it might not be a completly bad idea to drop this test. Perhaps it just needs splitting, to test only one thing per test. I'll add additional suggestions.

Reply:
Done https://github.com/neondatabase/neon/pull/5335/commits/0dc3cc030e39e3677c5fb56a2501a166aa4f465e